### PR TITLE
fixed issue #26288

### PR DESCRIPTION
**Pull request by shubhamkhandelwalrtpl**
### Description (*)

missing Throw in copy pasted code is fixed 
### Fixed Issues (if relevant)

1. magento/magento2#<26288>: missing Throw in copy pasted code 

### Manual testing scenarios (*)

There are no testing scenario. 

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by rogyar** [comment]
Hi @shubhamkhandelwalrtpl. Thank you for your collaboration. According to the [definition of done](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing_dod.html) all changes should be covered by automated tests. Could I kindly ask you to cover your fix wit a simple unit test, please? 

Thank you.

File: app/code/Magento/Customer/Model/ForgotPasswordToken/GetCustomerByToken.php [hunk 1]
```diff
@@ -73,7 +73,7 @@ public function execute(string $resetPasswordToken):CustomerInterface
         }
         if ($found->getTotalCount() === 0) {
             //Customer with such token not found.
```
Comment (lbajsarowicz) [inline app/code/Magento/Customer/Model/ForgotPasswordToken/GetCustomerByToken.php]: Actually, you can remove the comments. Code is self-explanatory.

File: app/code/Magento/Customer/Model/ForgotPasswordToken/GetCustomerByToken.php [hunk 2]
```diff
@@ -73,7 +73,7 @@ public function execute(string $resetPasswordToken):CustomerInterface
...
-            new NoSuchEntityException(
+           throw new NoSuchEntityException(
                 new Phrase(
```
Comment (lbajsarowicz) [inline app/code/Magento/Customer/Model/ForgotPasswordToken/GetCustomerByToken.php]: You can replace new Phrase with `__(...)`

**Review by lbajsarowicz — CHANGES_REQUESTED** [review]
Could you also change `return $found->getItems()[0]` with more reliable replacement like `return current($found->getItems())`? This is more independent from array keys.

**Comment by engcom-Charlie** [comment]
Hi @shubhamkhandelwalrtpl, please sign CLA.

File: app/code/Magento/Customer/Model/ForgotPasswordToken/GetCustomerByToken.php [hunk 3]
```diff
@@ -71,16 +71,9 @@ public function execute(string $resetPasswordToken):CustomerInterface
...
-            //Customer with such token not found.
-           throw new NoSuchEntityException(
-                new Phrase(
-                    'No such entity with rp_token = %value',
-                    [
-                        'value' => $resetPasswordToken
-                    ]
-                )
-            );
+            throw new NoSuchEntityException(__('No such entity with rp_token = %1', $resetPasswordToken));
```
Comment (lbajsarowicz) [inline app/code/Magento/Customer/Model/ForgotPasswordToken/GetCustomerByToken.php:76]: You can use placeholders in such situaltions:
```php
throw new NoSuchEntityException(
    __('No such entity with rp_token = %token', ['token' => $resetPasswordToken])
);
```

**Review by lbajsarowicz — APPROVED** [review]
Looks OK :-) 
Just minor improvement for readability.

**Comment by magento-engcom-team** [comment]
Hi @lbajsarowicz, thank you for the review.
[ENGCOM-6769](https://jira.corp.magento.com/browse/ENGCOM-6769) has been created to process this Pull Request

**Comment by engcom-Bravo** [comment]
:heavy_check_mark: QA Passed

The test is added and passes
forg_pass_token1

**Review by lbajsarowicz — CHANGES_REQUESTED** [review]
Sign license, please.

**Review by rogyar — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @rogyar, thank you for the review.
[ENGCOM-6769](https://jira.corp.magento.com/browse/ENGCOM-6769) has been created to process this Pull Request

**Comment by engcom-Bravo** [comment]
:heavy_check_mark: QA Passed

**Comment by engcom-Golf** [comment]
HI @shubhamkhandelwalrtpl we cannot process this PR unit you not signed CLA.

**Comment by lenaorobei** [comment]
Closing this PR since CLA is not signed.
@shubhamkhandelwalrtpl please feel free to reopen if you are going to finish this PR. Thanks.

**Pull request closed without merging**